haverhill_311_function/app.py

```diff
@@ -16,9 +16,22 @@ def lambda_handler(event, context):
     context: object, required
         Lambda Context runtime methods and attributes
     """
-    data = qalert.pull_data_test()
+    data = qalert.pull()
+    if not data:
+        return
+
+    latest_date = data[0]["createDate"]
     qalert_requests: List[db.QAlertRequest] = sanitizer.sanitize(
         qalert_data=data
     )
     with db.QAlertDB() as qalert_db:
-        qalert_db.save_many(requests=qalert_requests)
+        try:
+            qalert_db.save_many(requests=qalert_requests)
+        except Exception as e:
+            print(e)
```

Comment:
if it is the case that an exception happens here, by catching it and not re-raising, we allow execution to continue. i don't think that's what we want in this case though.